canvas_sdk/data/task.py

```diff
@@ -0,0 +1,51 @@
+from datetime import datetime
+from enum import Enum
+
+from canvas_sdk.data import Model
+from canvas_sdk.data.patient import Patient
+from canvas_sdk.data.staff import Staff
+from canvas_sdk.effects import Effect, EffectType
+
+
+class Task(Model):
+    class Status(Enum):
+        COMPLETED = "Completed"
+        CLOSED = "Closed"
+        OPEN = "Open"
+
+    id: str | None = None
+    assignee: Staff | None = None
+    patient: Patient | None = None
+    title: str | None = None
+    due: datetime | None = None
+    status: Status | None = None
+    comments: "list[TaskComment] | None" = None
+    labels: list[str] | None = None
+
+    def create(self) -> Effect:
+        return Effect(type=EffectType.CREATE_TASK, payload=self.model_dump_json_nested())
+
+    def update(self) -> Effect:
+        if not self.id:
+            raise ValueError("Cannot update a Task without an id")
+        payload = self.model_dump_json_nested(exclude_unset=True)
+        return Effect(type=EffectType.UPDATE_TASK, payload=payload)
+
+    def add_comment(self, comment: str) -> Effect:
```

Comment:
Since this isn't actually adding a comment, and is rather creating an effect that will add a comment, I wonder if this method actually belongs on TaskComment instead with a name like "create" (though I definitely get the initial desire to put it on Task).

Reply:
I was thinking for simplicity's sake that this might be easier for plugin devs.

Reply:
@csande If you compare two (nonfunctional, imaginary) plugins, one which comments on a task with `task.add_comment()` and the other with `task_comment.create()`, I think the ergonomics of `add_comment` are much better.